utils/src/main/kotlin/Utils.kt

```diff
@@ -221,6 +222,29 @@ fun getOrtDataDirectory() =
         File(it)
     } ?: getUserHomeDirectory().resolve(".ort")
 
+/**
+ * Migrate locl scan results from previously used directories to the current `~/.ort/results` directory.
+ */
+fun Any.migrateLocalFileScanStorage() {
+    fun hardlinkOldToNew(oldRoot: File, newRoot: File) {
+        oldRoot.walkBottomUp().filter { it.isFile }.forEach { oldFile ->
+            val newFile = newRoot.resolve(oldFile.relativeTo(oldRoot))
+            if (!newFile.exists()) {
+                runCatching {
+                    newFile.parentFile.safeMkdirs()
+                    Files.createLink(oldFile.toPath(), newFile.toPath())
+                }.onSuccess {
+                    log.debug { "Hard-linked '$oldFile' to '$newFile'." }
+                }
+            }
+        }
+    }
+
+    val newRoot = getOrtDataDirectory().resolve("results")
+    hardlinkOldToNew(getOrtDataDirectory().resolve("scanner/results"), newRoot)
```

Comment:
This should not be done, if the old path (or any other path) is set in the configuration file.